controllers/networking-calico/charts/internal/calico/templates/calico.yaml

```diff
@@ -334,43 +384,120 @@ spec:
             value: "true"
           # Uncomment these lines to enable prometheus metrics.  Since Typha is host-networked,
           # this opens a port on the host, which may need to be secured.
-          #- name: TYPHA_PROMETHEUSMETRICSENABLED
-          #  value: "true"
-          #- name: TYPHA_PROMETHEUSMETRICSPORT
-          #  value: "9093"
+#          - name: TYPHA_PROMETHEUSMETRICSENABLED
+#            value: "true"
+#          - name: TYPHA_PROMETHEUSMETRICSPORT
+#            value: "9093"
         livenessProbe:
-          exec:
-            command:
-            - calico-typha
-            - check
-            - liveness
+          httpGet:
+            path: /liveness
+            port: 9098
+            host: localhost
           periodSeconds: 30
           initialDelaySeconds: 30
         readinessProbe:
-          exec:
-            command:
-            - calico-typha
-            - check
-            - readiness
+          httpGet:
+            path: /readiness
+            port: 9098
+            host: localhost
           periodSeconds: 10
-
-
 ---
-
+apiVersion: {{ include "deploymentversion" . }}
+kind: Deployment
+metadata:
+  name: calico-typha-horizontal-autoscaler
+  namespace: kube-system
+  annotations:
+    checksum/configmap-calico: {{ include (print $.Template.BasePath "/config.yaml") . | sha256sum }}
+  labels:
+    k8s-app: calico-typha-autoscaler
+    kubernetes.io/cluster-service: "true"
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      k8s-app: calico-typha-autoscaler
+  template:
+    metadata:
+      labels:
+        k8s-app: calico-typha-autoscaler
+    spec:
+      priorityClassName: system-cluster-critical
+      securityContext:
+        supplementalGroups: [ 65534 ]
+        fsGroup: 65534
+      containers:
+        - image: {{ index .Values.images "typha-cpa" }}
+          name: autoscaler
+          command:
+            - /cluster-proportional-autoscaler
+            - --namespace=kube-system
+            - --configmap=calico-typha-horizontal-autoscaler
+            - --target=deployment/calico-typha
+            - --logtostderr=true
+            - --v=2
+          resources:
+            requests:
+              cpu: 10m
+            limits:
+              cpu: 10m
+      serviceAccountName: typha-cpha
+---
+kind: Deployment
+apiVersion: {{ include "deploymentversion" . }}
+metadata:
+  name: calico-typha-vertical-autoscaler
+  namespace: kube-system
+  annotations:
+    checksum/configmap-calico: {{ include (print $.Template.BasePath "/config.yaml") . | sha256sum }}
```

Comment:
The annotation needs to be part of the pod template and not part of the deployment metadata.
Also, I would separate the two configs into dedicated files, because right now the checksum you are computing takes into account both the calico-config as well as the autoscaler config. You don't want the autoscaler to restart if the calico config changes and vice versa.